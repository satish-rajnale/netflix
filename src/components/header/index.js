import { useState } from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Profile,
  PlayButton,
  Picture,
  Dropdown,
  Group,
  Link,
  FeatureCallOut,
  Text,
  Background,
  ButtonLink,
  Container,
  Logo,
  Feature,
  SearchInput,
  SearchIcon,
  Search,
} from "./styles/header";
import React from 'react';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((searchActive) => !searchActive)}
      >
        <img src="/images/icons/search.png" alt="search" />
      </SearchIcon>
      <SearchInput
        placeholder="Search films & series"
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        active={searchActive}
      />
    </Search>
  );
};
