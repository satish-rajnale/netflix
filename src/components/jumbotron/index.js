import React from 'react';
import {Inner} from './styles/jumbotron'

export default function Jumbotron({children, direction = 'row', ...restProps}){
    return (
        <Inner direction={direction}>
            <p>Hello again</p>
        </Inner>
    )
}


Jumbotron.Container = function JumbotronContainer({ children, ...restProps}){
    return <Container {...restProps}>{children}</Container>
}