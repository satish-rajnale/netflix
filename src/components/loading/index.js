import { LockBody, Spinner, Picture, ReleaseBody } from "./styles/loading";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingReleasebody() {
  return <ReleaseBody />;
};
