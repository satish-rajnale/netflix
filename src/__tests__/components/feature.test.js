import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { Feature } from "../../components";

describe("<Feature/>", () => {
    it("renders the <Feature/> with data", () => {
        const { container, getByText} = render(
            <Feature>
                <Feature.Title>
                    Unlimited Films, TV programmes & more.
                </Feature.Title>
                <Feature.SubTitle>
                    Watch anywhere. Cancel at anytime.
                </Feature.SubTitle>
            </Feature>
        );


        expect(getByText("Unlimited Films, TV programmes & more.")).toBeTruthy();
        expect(getByText("Watch anywhere. Cancel at anytime.")).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });


    it("renders the <Feature/> only title", () => {
        const { container, getByText, queryByText} = render(
            <Feature>
                <Feature.Title>
                    Unlimited Films, TV programmes & more.
                </Feature.Title>
            </Feature>
        );


        expect(getByText("Unlimited Films, TV programmes & more.")).toBeTruthy();
        expect(queryByText("Watch anywhere. Cancel at anytime.")).toBeFalsy();

        expect(container.firstChild).toMatchSnapshot();
    })
})