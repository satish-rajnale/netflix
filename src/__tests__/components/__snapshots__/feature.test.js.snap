// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature/> renders the <Feature/> only title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 isjLws"
>
  <h1
    class="feature__Title-sc-6swrp6-1 biQKvD"
  >
    Unlimited Films, TV programmes & more.
  </h1>
</div>
`;

exports[`<Feature/> renders the <Feature/> with data 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 isjLws"
>
  <h1
    class="feature__Title-sc-6swrp6-1 biQKvD"
  >
    Unlimited Films, TV programmes & more.
  </h1>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 bONLfr"
  >
    Watch anywhere. Cancel at anytime.
  </h2>
</div>
`;
